packages/generate/src/syntax/path.ts

```diff
@@ -39,7 +41,17 @@ type getChildOfObjectTypeSet<
   Root extends ObjectTypeSet,
   ChildKey extends keyof Root["__element__"]["__pointers__"],
 > = TypeSet<
-  Root["__element__"]["__pointers__"][ChildKey]["target"],
+  ChildKey extends "name"
+    ? Root extends { [typenameSymbol]: string }
+      ? ScalarType<
+          "std::str",
+          TypesystemOptions["future"]["strictTypeNames"] extends true
+            ? Root[typeof typenameSymbol]
+            : string,
+          Root[typeof typenameSymbol]
+        >
```

Comment:
I'm not sure I understand the difference between Scalar's `TsType` & `TsArgType`. I was thinking that they were the output / input types respectively.

I don't understand why the input type would be called out here explicitly and set to the literal string - where/how would that be used be as an input?

I figured the output type remained as a `string` for BC reasons. So I created another future option to make it strict.
This is for this btw:
```ts
scope => ({ t: scope.__type__.name })
```

But then I was thinking about this case
```ts
scope => ({ __type__: { name: true } })
```
Which had already made the output type a strict string.

So I thought the other case was a miss, but there is some explicit code here.

I guess all that to say I could just do this:
```suggestion
      ? ScalarType<"std::str", Root[typeof typenameSymbol]>
```

Reply:
@scotttrinh thoughts here? I don't think we need a flag for this & just use my suggestion

Reply:
Good question. I'm not sure, but I'm fine with going with the suggested change for now and looking at what the fallout is.